darts/tests/models/forecasting/test_regression_ensemble_model.py

```diff
@@ -363,3 +363,228 @@ def test_call_backtest_regression_ensemble_local_models(self):
         # -10 comes from the maximum minimum train series length of all models
         assert ensemble.extreme_lags == (-10 - regr_train_n, 0, None, None, None, None)
         ensemble.backtest(series)
+
+    def test_stochastic_regression_ensemble_model(self):
+        quantiles = [0.25, 0.5, 0.75]
+
+        # probabilistic ensembling model
+        linreg_prob = LinearRegressionModel(
+            quantiles=quantiles, lags_future_covariates=[0], likelihood="quantile"
+        )
+
+        # deterministic ensembling model
+        linreg_dete = LinearRegressionModel(lags_future_covariates=[0])
+
+        # every models are probabilistic
+        ensemble_allproba = RegressionEnsembleModel(
+            forecasting_models=[
+                self.get_probabilistic_global_model([-1, -3], quantiles),
+                self.get_probabilistic_global_model([-2, -4], quantiles),
+            ],
+            regression_train_n_points=10,
+            regression_model=linreg_prob.untrained_model(),
+        )
+
+        self.assertTrue(ensemble_allproba._models_are_probabilistic())
+        self.assertTrue(ensemble_allproba._is_probabilistic())
+        ensemble_allproba.fit(self.ts_random_walk[:100])
+        # probabilistic forecasting is supported
+        pred = ensemble_allproba.predict(5, num_samples=10)
+        self.assertEqual(pred.n_samples, 10)
+
+        # forecasting models are a mix of probabilistic and deterministic, probabilistic regressor
+        ensemble_mixproba = RegressionEnsembleModel(
+            forecasting_models=[
+                self.get_probabilistic_global_model([-1, -3], quantiles),
```

Comment:
can you check also mixed proba model with train_num_samples > 1 and a reduction?

Reply:
It fails because it cannot stack `TimeSeries` on the component axis if they don't have the same number of samples. We can either add an exception to prevent this scenario or apply the reduction inside `_make_multiple_prediction`. WDYT?

Reply:
had to move the reduction to `_make_multiple_prediction()` so that they could be stacked properly